### PR TITLE
[SPARK-36535][SQL] Refine the sql reference doc

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Refine the SQL reference doc:
- remove useless subitems in the sidebar
- remove useless sub-menu-pages (e.g. `sql-ref-syntax-aux.md`)
- avoid using `#####` in `sql-ref-literals.md`

### Why are the changes needed?

The subitems in the sidebar are quite useless, as the menu page serves the same functionalities:
WX20210817-235840@2x
It's also extra work to keep the manu page and sidebar subitems in sync (The ANSI compliance page is already out of sync).

The sub-menu-pages are only referenced by the sidebar, and duplicates the content of the menu page. As a result, the `sql-ref-syntax-aux.md` is already outdated compared to the menu page. It's easier to just look at the menu page.

The `#####` is not rendered properly:
WX20210818-000119@2x
It's better to avoid using it.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

N/A

### CODE REVIEW

File: docs/_data/menu-sql.yaml [hunk 1]
```diff
@@ -75,28 +75,12 @@
   subitems:
     - text: ANSI Compliance
       url: sql-ref-ansi-compliance.html
-      subitems:
```
Comment (cloud-fan) [inline docs/_data/menu-sql.yaml:78]: This is outdated.

File: docs/sql-ref-syntax-aux.md [hunk 2]
```diff
@@ -1,29 +0,0 @@
...
-  Unless required by applicable law or agreed to in writing, software
-  distributed under the License is distributed on an "AS IS" BASIS,
-  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-  See the License for the specific language governing permissions and
-  limitations under the License.
----
-
-Besides the major SQL statements such as Data Definition Statements, Data Manipulation Statements and Data Retrieval Statements, Spark SQL also supports the following Auxiliary Statements:
-
- * [ANALYZE](sql-ref-syntax-aux-analyze.html)
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-aux.md:24]: This list is outdated.

**Comment by cloud-fan** [comment]
cc @gengliangwang @dongjoon-hyun

File: docs/_data/menu-sql.yaml (hunk 1 above)
Comment (gengliangwang) [reply docs/_data/menu-sql.yaml:78]: Nice catch!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. New doc looks better to me.

Screen Shot 2021-08-17 at 12 45 21 PM

**Comment by dongjoon-hyun** [comment]
Merged to master/3.2.

**Pull request closed without merging**